src/renderer/intl_provider.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {IntlProvider as BaseIntlProvider} from 'react-intl';
+
+import {GET_LANGUAGE_INFORMATION, RETRIEVED_LANGUAGE_INFORMATION} from 'common/communication';
+
+import {Language} from '../../i18n/i18n';
+
+type State = {
+    language?: Language;
+}
+
+export function t(id: string) {
+    return id;
+}
+
+export default class IntlProvider extends React.PureComponent<any, State> {
+    constructor(props: any) {
+        super(props);
+        this.state = {};
+    }
+
+    componentDidMount() {
+        window.addEventListener('message', this.handleMessageEvent);
```

Comment:
This should be removed on unmount since it seems like we're going to have multiple IntlProviders mounted at different points. Otherwise, we're leaking these